datapipelines/src/main/kotlin/io/customer/sdk/DataPipelineInstance.kt

```diff
@@ -127,12 +128,14 @@ abstract class DataPipelineInstance : CustomerIOInstance {
      * @param name Name of the action
      * @param properties to describe the action. Needs to be [serializable](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serializers.md)
      * @param serializationStrategy strategy to serialize [properties]
+     * @param enrichment closure to enrich the generated event
      * @see [Learn more](https://customer.io/docs/cdp/sources/source-spec/track-spec/)
      */
     abstract fun <T> track(
         name: String,
         properties: T,
-        serializationStrategy: SerializationStrategy<T>
+        serializationStrategy: SerializationStrategy<T>,
+        enrichment: EnrichmentClosure? = null
```

Comment:
I think this method can't be used directly from Java, so we probably don't need to add `@JvmOverloads`. However, since it's public, just wanted to double-check if you share the same thoughts, just to make sure we don't break any public APIs unknowingly.